src/components/controls/link/km-link.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <div  class="scbd-common km-link">
+        <button type="button" class="btn btn-primary" @click="addLink()">
+            <slot name="link-button-label">+ Add Link</slot>
+        </button>  
+        <select-file-button  @on-file-selected="receiveFile"> <slot name="file-button-label">+ Add Files</slot></select-file-button> 
+             
+       <file-upload-editor ref="fileEditorRef" @on-close="onFileUploadEditorClose">
+            <template v-slot:modalTitle>
+            <slot name="link-dialog-title"   >        
+                File Upload   
+            </slot>
+            </template>       
+        </file-upload-editor>
+
+        <link-editor ref="linkEditorRef" @on-close="onLinkEditorClose">       
+        <template v-slot:modalTitle>
+            <slot name="link-dialog-title"   >        
+            Edit link    
+            </slot>
+        </template>       
+        </link-editor>
+    
+      <km-view-links v-model="links"   @on-delete ="removeLink($event)"   @on-edit = "editLink($event)"></km-view-links>  
+    
+    </div>
+</template>
+<script setup>
+    import { ref ,shallowRef} from 'vue'  
+    import kmViewLinks from './km-view-links.vue';
+    import linkEditor  from './link-editor.vue';
+    import fileUploadEditor  from './file-upload-editor.vue';
+    import selectFileButton from '../../inputs/select-file-button.vue'
+
+    const links = defineModel({type:Array, required:true, default:[]});
+    const linkEditorRef= shallowRef(null); 
```

Comment:
add two properties, `allowLinks: default true`, `allowFiles: default true`, hide links and file based on the props.

Reply:
add prop for `mineTypes`

Reply:
done